services/relayer/internal/msgobserver/eth/observer/observer_impl_v1.go

```diff
@@ -145,25 +146,33 @@ func (o *ObserverImplV1) StopObserve() error {
 
 // routine is the observe routine.
 func (o *ObserverImplV1) routine(chainID *big.Int, chainAP string, addr common.Address, end chan bool) {
-	chain, err := ethclient.Dial(chainAP)
-	if err != nil {
-		logging.Error(err.Error())
-		return
-	}
-	defer chain.Close()
+	for {
+		chain, err := ethclient.Dial(chainAP)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		defer chain.Close()
 
-	sfc, err := functioncall.NewSfc(addr, chain)
-	if err != nil {
-		logging.Error(err.Error())
-		return
-	}
+		sfc, err := functioncall.NewSfc(addr, chain)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
 
-	observer, err := NewSFCBridgeObserver(chainID.String(), addr.String(), sfc, o.mq, end)
-	if err != nil {
-		logging.Error(err.Error())
-		return
+		observer, err := NewSFCBridgeObserver(chainID.String(), addr.String(), sfc, o.mq, end)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		if observer.Start() == nil {
+			break
+		} else {
+			logging.Warn("Error in observing event. Retry in 3 seconds...")
+			chain.Close()
+			time.Sleep(3 * time.Second)
```

Comment:
should we have a finite number of retries?

Reply:
At the moment, it fails if the retry failed. So say we have a network loss, we will wait 3 seconds and try to dial again and if the retry failed, the program will exit at line 153.